runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -0,0 +1,2574 @@
+   }*/
+bool
+TR_J9ServerVM::isPrimitiveArray(TR_OpaqueClassBlock *clazz)
+   {
+   uint32_t modifiers = 0;
+   TR_OpaqueClassBlock * componentClass = NULL;
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   JITServerHelpers::getAndCacheRAMClassInfo((J9Class *)clazz, _compInfoPT->getClientData(), stream, JITServerHelpers::CLASSINFO_ROMCLASS_MODIFIERS, (void *)&modifiers, JITServerHelpers::CLASSINFO_COMPONENT_CLASS, (void *)&componentClass);
+
+   if (!J9_ARE_ALL_BITS_SET(modifiers, J9AccClassArray))
+      return false;
+
+   JITServerHelpers::getAndCacheRAMClassInfo((J9Class *)componentClass, _compInfoPT->getClientData(), stream, JITServerHelpers::CLASSINFO_ROMCLASS_MODIFIERS, (void *)&modifiers);
+   return J9_ARE_ALL_BITS_SET(modifiers, J9AccClassInternalPrimitiveType) ? true : false;
+
+   }
+
+uint32_t
+TR_J9ServerVM::getAllocationSize(TR::StaticSymbol *classSym, TR_OpaqueClassBlock *clazz)
+   {
+   uintptrj_t totalInstanceSize = 0;
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   JITServerHelpers::getAndCacheRAMClassInfo((J9Class *)clazz, _compInfoPT->getClientData(), stream, JITServerHelpers::CLASSINFO_TOTAL_INSTANCE_SIZE, (void *)&totalInstanceSize);
+
+   uint32_t objectSize = sizeof(J9Object) + (uint32_t)totalInstanceSize;
+   return ((objectSize >= J9_GC_MINIMUM_OBJECT_SIZE) ? objectSize : J9_GC_MINIMUM_OBJECT_SIZE);
+   }
+
+TR_OpaqueClassBlock *
+TR_J9ServerVM::getObjectClass(uintptrj_t objectPointer)
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getObjectClass, objectPointer);
+   return std::get<0>(stream->read<TR_OpaqueClassBlock *>());
+   }
+
+uintptrj_t
+TR_J9ServerVM::getStaticReferenceFieldAtAddress(uintptrj_t fieldAddress)
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getStaticReferenceFieldAtAddress, fieldAddress);
+   return std::get<0>(stream->read<uintptrj_t>());
+   }
+
+bool
+TR_J9ServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *clazz)
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, method, clazz);
+   return std::get<0>(stream->read<bool>());
+   }
+
+bool
+TR_J9ServerVM::hasFinalFieldsInClass(TR_OpaqueClassBlock *clazz)
+   {
+   bool classHasFinalFields = false;
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   JITServerHelpers::getAndCacheRAMClassInfo((J9Class *)clazz, _compInfoPT->getClientData(), stream, JITServerHelpers::CLASSINFO_CLASS_HAS_FINAL_FIELDS, (void *)&classHasFinalFields);
+
+   return classHasFinalFields;
+   }
+
+const char *
+TR_J9ServerVM::sampleSignature(TR_OpaqueMethodBlock * aMethod, char *buf, int32_t bufLen, TR_Memory *trMemory_unused)
```

Comment:
We should be able to avoid a message here, but lately this message no longer appears.
I'll open an issue